daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/unparsers/UState.scala

```diff
@@ -78,6 +79,9 @@ abstract class UState(
   extends ParseOrUnparseState(vbox, diagnosticsArg, dataProcArg, tunable)
   with Cursor[InfosetAccessor] with ThrowsSDE with SavesErrorsAndWarnings {
 
+  def pushTRD(rd: RuntimeData): Unit
+  def popTRD(rd: RuntimeData): RuntimeData
```

Comment:
doc  that the rd is passed to popTRD so we verify we got what we expected to pop.